File: clients/admin/src/mutations/ChangeUserPassword.ts
import client from '../modules/apollo';
import mutationPromise from '../helpers/mutationPromise';
import gql from 'graphql-tag';

const mutation = gql`
  mutation ChangeUserPasswordMutation($input: ChangeUserPasswordInput!) {
    changeUserPassword(input: $input) {
      success
      message
      userErrors {
        field
        message
      }
    }
  }
`;

function commit(variables: ChangeUserPasswordMutationVariables) {
  return mutationPromise((resolve, reject) => {
    client
      .mutate({
        mutation,
        variables
      })
      .then(({ data: { changeUserPassword } }) => {
        resolve(changeUserPassword as MutationResult);
      })
      .catch(error => {
        reject(error);
      });
  });
}

export default { commit };

File: app/graphql/mutations/change_user_password.rb
class Mutations::ChangeUserPassword < Mutations::BaseMutation
  graphql_name "ChangeUserPassword"

  argument :input, Inputs::ChangeUserPasswordInput, required: true

  field :user, Types::User, null: false
  field :success, Boolean, null: false
  field :message, String, null: true
  field :userErrors, [Types::Error], null: true
end
